kolibri/plugins/coach/assets/src/composables/usePreviousRoute.js

```diff
@@ -0,0 +1,67 @@
+import { ref, provide, inject } from 'vue';
+import { onBeforeRouteUpdate, useRouter } from 'vue-router/composables';
+
+/**
+ * Composable for providing a route tracking context to track the previous route. You can
```

Comment:
Does this composable makes sense? The benefit of having this like this is that if I tracked the routes higher in the routing hierarchy as for example in the coach app.js, then I would need to validate if the previous route belongs to the routes related to the side panel and that seems like adding a little bit more complexity.

Reply:
It makes sense to me be used like this where it's scoped to a particular sub-hierarchy in the code. It does seem like it's something that is specific to the side panel use case - do you anticipate this being used in other contexts as well?

Reply:
For now, yes, just for side panels, but it would work on any sub-route context with scoped navigation